app/src/main/java/com/isc/hermes/controller/IncidentFormController.java

```diff
@@ -105,19 +126,71 @@ private void handleUploadResponse(Integer responseCode) {
      * </p>
      */
     public void setIncidentComponents() {
-        Spinner incidentType = ((AppCompatActivity) context).findViewById(R.id.incident_spinner);
-        ArrayAdapter<CharSequence> adapter=ArrayAdapter.createFromResource(context, R.array.incidents_type, R.layout.incident_spinner_items);
-        adapter.setDropDownViewResource(android.R.layout.simple_spinner_item);
-        incidentType.setAdapter(adapter);
+        String[] incidentTypes = context.getResources().getStringArray(R.array.incidents_type);
+        String[] incidentTypeColors = context.getResources().getStringArray(R.array.incidents_type_colors);
+        String[] incidentTypeIcons = context.getResources().getStringArray(R.array.incidents_type_icons);
+
+        if (incidentTypes.length == incidentTypeColors.length &&
+                incidentTypeIcons.length == incidentTypes.length) {
+            for (int i = 0; i < incidentTypes.length; i++) {
+                IncidentTypeButton incidentTypeButton = new IncidentTypeButton();
+                Button button = incidentTypeButton.getIncidentTypeButton(
+                        context,
+                        incidentTypes[i].toLowerCase(),
+                        Color.parseColor((String) incidentTypeColors[i]),
+                        incidentTypeIcons[i]);
+                setIncidentButtonAction(button);
+
+                incidentTypesContainer.addView(button);
+            }
+        } else {
+            Timber.i("The arrays about colors, icons and the type buttons have a different size");
+        }
```

Comment:
Add this message to the string.xml file, and call it from here.

Reply:
Done